honeybee_radiance_folder/folderutil.py

```diff
@@ -0,0 +1,314 @@
+# -*- coding: utf-8 -*-
+"""Utilities for Radiance folder structure."""
+import os
+import json
+
+
+class GeometryState(object):
+    """A state for a dynamic geometry from Radiance files.
+    
+    Attributes:
+        name
+        default
+        direct
+    """
+    def __init__(self, name, default, direct):
+        """Geometry State.
+
+        Args:
+            name: Name for state.
+            default: Path to file to be used for normal representation of the geometry.
+            direct: Path to file to be used for direct studies.
+        """
+        self.name = name
+        self.default = default
+        self.direct = direct
+
+    @classmethod
+    def from_dict(cls, input_dict):
+        """Create a state from an input dictionary.
+        ```
+        {
+          "name": "grass_covered",
+          "default": "ground..summer..000.rad",
+          "direct": "ground..direct..000.rad",
+        }
+        ```
+        """
+        # TODO: add checks for keys
+        name = input_dict['name']
+        default = input_dict['default']
+        direct = input_dict['direct']
+        return cls(name, default, direct)
+
+    def validate(self, folder):
+        """Validate files in this state."""
+        assert os.path.isfile(os.path.join(folder, self.default)), \
+            'Failed to find default file for %s' % self.name
+        assert os.path.isfile(os.path.join(folder, self.direct)), \
+            'Failed to find direct file for %s' % self.name
+
+    def __repr__(self):
+        return 'GeometryState: {}'.format(self.name)
+
+
+class ApertureState(GeometryState):
+    """A state for a dynamic aperture from Radiance files.
+    
+    Attributes:
+        name
+        default
+        direct
+        black
+        tmtx
+        vmtx
+        dmtx
+    """
+
+    def __init__(self, name, default, direct,
+            black=None, tmtx=None, vmtx=None, dmtx=None):
+        """Aperture State.
+
+        Args:
+            name: Name for state.
+            default: Path to file to be used for normal representation of the geometry.
+            direct: Path to file to be used for direct studies.
+            black: Path to file for blacking out the window.
+            tmtx: Path to file for transmittance matrix.
+            vmtx: Path to file for transmittance matrix.
+            dmtx: Path to file for transmittance matrix.
+        """
+        GeometryState.__init__(self, name, default, direct)
+        self.black = black
+        self.tmtx = tmtx
+        self.vmtx = vmtx
+        self.dmtx = dmtx
```

Comment:
Is there a reason why you don't check whether the input types are correct?  Or should that be in a TODO here?
The same thing goes for the other classes here.

Reply:
The only check I can do here is to see if it is string or None. Since in %90 of cases the values are coming from a JSON file the validation happens there and is set to True by default.

https://github.com/ladybug-tools/honeybee-radiance-folder/pull/1/files#diff-21cfd4ce4540b1619d197c88573e239fR282